### ISSUE TITLE
OpenCV does not look in lib64 directories for dependencies

### ISSUE THREAD

**Issue by mcneish1** [spack/spack#8160]
Recently (#8102), libjpeg-turbo started building libraries to lib64 instead of lib on my machine.

I believe this is the culprit (in opencv's package.py):
```
        if '+jpeg' in spec:
            libjpeg = spec['jpeg']
            args.extend([
                '-DBUILD_JPEG:BOOL=OFF',
                '-DJPEG_LIBRARY:FILEPATH={0}'.format(
                    join_path(libjpeg.prefix.lib,
                              'libjpeg.{0}'.format(dso_suffix))),
                '-DJPEG_INCLUDE_DIR:PATH={0}'.format(libjpeg.prefix.include)
            ])
```

This does not consider the possibility that the path should be `libjpeg.prefix.lib64`.

I'm currently manually symlinking lib64/libjpeg.so to lib/libjpeg.so, as a stopgap.

 * Does `libjpeg.prefix.lib64` exist, in the context above? Can it? Should it?
 * Is there some way to get libjpeg-turbo to install to lib instead of lib64? None of the changes in libjpeg-turbo stood out as the reason it changed directories.

**Comment by davydden** [comment]
you should add `libs()` to `libjpeg-turbo` which should be responsible for knowing where its libs are.

**Comment by mcneish1** [comment]
So:
```
    @property
    def libs(self):
        return find_libraries("libjpeg*", root=self.prefix, recursive=True)
```
works, and `spec['libjpeg-turbo'].libs` didn't work otherwise. Does that look right to you?

On the opencv side, a lot of the dependencies hardcode their paths as lib[x].prefix.lib, which means a similar build change in any of the other dependencies could take opencv out again.

Question: `spec['x'].libs`, and `find_libraries` seem to indicate that multiple returns are possible, however `find_libraries` in libjpeg-turbo only returned the one string, and everything in opencv needs one library, not multiple. What's the right solution here? I'm guessing it isn't:
```
libs = spec['x'].libs
if isinstance(libs, list):
    libs = libs[0]
```

**Comment by scheibelp** [comment]
That's currently what you have to do: it's slightly inconvenient when you want to refer to (and expect exactly) one library file. `.libs` will always return a list though (unless it is redefined by the `libjpeg-turbo` package in which case the override should return a sequence also) so you can say:

`lib = spec['jpeg'].libs[0]`

Does `opencv` specifically require `libjpeg-turbo` vs. `jpeg`?

**Comment by mcneish1** [comment]
>the override should return a sequence also
I'm returning straight from find_libraries, it doesn't appear to be a list, checking that now.

>lib = spec['jpeg'].libs[0]
I'll do that

>Does opencv specifically require libjpeg-turbo vs. jpeg?
No, just jpeg. I'll test the changes against the regular version of libjpeg. `libs` in libjpeg-turbo is broken either way, though.

New question:
In opencv, the dependencies are currently specified as
`join_path(libx.prefix.lib, 'libx.{0}'.format(dso_suffix))`
Am I right in assuming `spec['x'].libs[0]` will handle the suffix/path joining correctly (as in, by just giving the full path to the dylib/so)?

**Comment by scheibelp** [comment]
> Am I right in assuming spec['x'].libs[0] will handle the suffix/path joining correctly (as in, by just giving the full path to the dylib/so

Yes: the paths returned by `spec['jpeg'].libs` are absolute (pretty sure, it's a bug if they aren't)

**Comment by mcneish1** [comment]
Resolved with #8179
